contracts/ETHMixer.sol

```diff
@@ -22,18 +22,20 @@ contract ETHMixer is Mixer {
   ) Mixer(_verifier, _denomination, _merkleTreeHeight, _operator) public {
   }
 
-  function _processDeposit() internal {
+  function _processDeposit() internal nonReentrant {
     require(msg.value == denomination, "Please send `mixDenomination` ETH along with transaction");
   }
 
-  function _processWithdraw(address payable _recipient, address payable _relayer, uint256 _fee, uint256 _refund) internal {
+  function _processWithdraw(address payable _recipient, address payable _relayer, uint256 _fee, uint256 _refund) internal nonReentrant {
     // sanity checks
     require(msg.value == 0, "Message value is supposed to be zero for ETH mixer");
     require(_refund == 0, "Refund value is supposed to be zero for ETH mixer");
 
-    _recipient.transfer(denomination - _fee);
+    (bool success, ) = _recipient.call.value(denomination - _fee)("");
```

Comment:
Now recipient is able to spend more than 2300 gas when processing incoming transfer, and this gas will be paid by relayer.  If this it not intentional, consider using “send” instead of “call”.

Reply:
https://diligence.consensys.net/blog/2019/09/stop-using-soliditys-transfer-now/

FYI using transfer() and send() is not safe.